test/e2e/checkpoint_test.go

```diff
@@ -1339,4 +1339,40 @@ var _ = Describe("Podman checkpoint", func() {
 		Expect(podmanTest.NumberOfContainersRunning()).To(Equal(0))
 	})
 
+	It("podman checkpoint and restore container with --file-locks", func() {
+		if !strings.Contains(podmanTest.OCIRuntime, "runc") {
+			// FIXME: Enable crun test when this feature has been released
+			// https://github.com/containers/crun/pull/783
+			Skip("Test only works with runc.")
+		}
+		localRunString := getRunString([]string{"--name", "test_name", ALPINE, "flock", "test.lock", "sleep", "100"})
+		session := podmanTest.Podman(localRunString)
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+
+		// Checkpoint is expected to fail without --file-locks
+		result := podmanTest.Podman([]string{"container", "checkpoint", "test_name"})
+		result.WaitWithDefaultTimeout()
+		Expect(result).Should(Exit(125))
```

Comment:
I'm fond of checking error messages, not just exit code. (The command could fail for other reasons, even a typo). Unfortunately criu does not emit useful diagnostics, but even just checking for the "blah blah -52" message would be better than nothing.

Reply:
Thanks, I've added a check for the error message.